tests/components/airvisual/test_config_flow.py

```diff
@@ -156,24 +189,74 @@ async def test_step_import(hass):
         }
 
 
-async def test_step_user(hass):
-    """Test that the user step works."""
-    conf = {
+async def test_step_node_pro(hass):
+    """Test the Node/Pro step."""
+    conf = {CONF_IP_ADDRESS: "192.168.1.100", CONF_PASSWORD: "my_password"}
+
+    with patch(
+        "homeassistant.components.airvisual.async_setup_entry", return_value=True
+    ), patch("pyairvisual.node.Node.from_samba"):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}, data={"type": "AirVisual Node/Pro"}
+        )
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input=conf
+        )
+        assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+        assert result["title"] == "Node/Pro (192.168.1.100)"
+        assert result["data"] == {
+            CONF_IP_ADDRESS: "192.168.1.100",
+            CONF_PASSWORD: "my_password",
+        }
+
+
+async def test_step_import(hass):
+    """Test the import step for both types of configuration."""
+    geography_conf = {
         CONF_API_KEY: "abcde12345",
-        CONF_LATITUDE: 32.87336,
-        CONF_LONGITUDE: -117.22743,
+        CONF_LATITUDE: 51.528308,
+        CONF_LONGITUDE: -0.3817765,
     }
 
     with patch(
         "homeassistant.components.airvisual.async_setup_entry", return_value=True
     ), patch("pyairvisual.api.API.nearest_city"):
         result = await hass.config_entries.flow.async_init(
-            DOMAIN, context={"source": SOURCE_USER}, data=conf
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=geography_conf
         )
+
         assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
-        assert result["title"] == "Cloud API (32.87336, -117.22743)"
+        assert result["title"] == "Cloud API (51.528308, -0.3817765)"
         assert result["data"] == {
             CONF_API_KEY: "abcde12345",
-            CONF_LATITUDE: 32.87336,
-            CONF_LONGITUDE: -117.22743,
+            CONF_LATITUDE: 51.528308,
+            CONF_LONGITUDE: -0.3817765,
         }
+
+
+async def test_step_user(hass):
+    """Test the user ("pick the integration type") step."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_USER},
+        data={"type": INTEGRATION_TYPE_GEOGRAPHY},
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "geography"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5403678)